sw/device/tests/sim_dv/i2c_device_tx_rx_test.c

```diff
@@ -185,12 +186,17 @@ bool test_main(void) {
     expected_data[i] = rand_testutils_gen32_range(0, 0xff);
   };
 
-  while (tx_empty_irq_seen == false) {
+  uint8_t tx_fifo_lvl;
+  CHECK_DIF_OK(dif_i2c_get_fifo_levels(&i2c, NULL, NULL, &tx_fifo_lvl, NULL));
+  // Write expected data to tx fifo.
+  while (tx_fifo_lvl > 0 && tx_empty_irq_seen == false) {
```

Comment:
I think it would be nice to add a timeout checking here just in case the fifo never gets empty.
```suggestion
  IBEX_SPIN_FOR (tx_fifo_lvl > 0 && tx_empty_irq_seen == false, <timeout>) {
```

Reply:
Nit/Question: Did you mean to poll the tx_fifo_lvl here?

Reply:
No, checking the `tx_empty_irq_seen` it's enough.

Reply:
> I think it would be nice to add a timeout checking here just in case the fifo never gets empty.

Thanks for the suggestion. I've made the change now.